services/blockstorage/service.go

```diff
@@ -82,8 +85,10 @@ func (s *service) GetLastCommittedBlockHeight(ctx context.Context, input *servic
 }
 
 func (s *service) CommitBlock(ctx context.Context, input *services.CommitBlockInput) (*services.CommitBlockOutput, error) {
+	logger := s.logger.WithTags(trace.LogFieldFrom(ctx))
+
 	txBlockHeader := input.BlockPair.TransactionsBlock.Header
```

Comment:
for me this can be a blocker. it feels unnatural to remember you need to create a logger with tags each time you start ...

Reply:
you could also send the context to the log when logging:
```golang
s.logger.info("foobar", trace.LogFieldFrom(ctx))
```